### ISSUE TITLE
not compatible with react-native `import jsonpatch from 'fast-json-patch' `

### ISSUE THREAD

**Issue by JasonMore** [Starcounter-Jack/JSON-Patch#169]
Hey guys, 

I was checking out your lib as an alternative to JSON8/patch my react-native project. I wasn't able to use it out of the box do to the lines below. `exports` is undefined, but `module.exports` is defined:

```
if (typeof exports !== "undefined") {
    exports.apply = jsonpatch.apply;
    exports.observe = jsonpatch.observe;
    exports.unobserve = jsonpatch.unobserve;
    exports.generate = jsonpatch.generate;
    exports.compare = jsonpatch.compare;
    exports.validate = jsonpatch.validate;
    exports.validator = jsonpatch.validator;
    exports.JsonPatchError = jsonpatch.JsonPatchError;
}
else {
    var exports = {};
    var isBrowser = true;
}
Object.defineProperty(exports, "__esModule", { value: true });
exports.default = jsonpatch;
/*
When in browser, setting `exports = {}`
fools other modules into thinking they're
running in a node environment, which breaks
some of them. Here is super light wieght fix.
*/
if (isBrowser) {
    exports = undefined;
}
```

I also use seamless-immutable which handles the same issue slightly different https://github.com/rtfeldman/seamless-immutable/blob/master/src/seamless-immutable.js#L739

```
  if (typeof define === 'function' && define.amd) {
    define(function() {
      return Immutable;
    });
  } else if (typeof module === "object") {
    module.exports = Immutable;
  } else if (typeof exports === "object") {
    exports.Immutable = Immutable;
  } else if (typeof window === "object") {
    window.Immutable = Immutable;
  } else if (typeof global === "object") {
    global.Immutable = Immutable;
```

I tried to make a PR but got stuck in typescript trying to define `module`, `exports` and `global`. But I did test that this change works in regular JS in my project. https://github.com/JasonMore/JSON-Patch/blob/master/src/json-patch-duplex.js#L635

**Comment by felixfbecker** [comment]
The export handling is extremely hacky. It should be a proper UMD wrapper: https://github.com/umdjs/umd

If TS can't find `module` etc. we just need to add `@types/node` as a dev dependency.

**Comment by alshakero** [comment]
This is fixed in 2.0.0. At least the exporting part. But I'm not 100% sure how this lib will behave under RN. Given RN gives you a strange `window` object.
